controller/src/main/java/org/jboss/as/controller/descriptions/DefaultResourceDescriptionProvider.java

```diff
@@ -99,11 +104,18 @@ public ModelNode getModelDescription(Locale locale) {
         }
 
         Set<? extends Capability> capabilities = registration.getCapabilities();
-        if (capabilities!=null&&!capabilities.isEmpty()){
+        if (capabilities !=null && !capabilities.isEmpty()){
             for (Capability capability: capabilities) {
                 ModelNode cap = result.get(ModelDescriptionConstants.CAPABILITIES).add();
                 cap.get(NAME).set(capability.getName());
                 cap.get(DYNAMIC).set(capability.isDynamicallyNamed());
+                if(capability.isDynamicallyNamed()) {
+                    String dynamicPart = capability.getDynamicName(createAliasPathAddress(pa)).substring(capability.getName().length());
+                    StringTokenizer tokenizer = new StringTokenizer(dynamicPart, ".$", false);
```

Comment:
If you use String.split you can get an array and again test for the case of a 1 element array whose element is the same as pa.getLastElement().getKey(). Don't output more description data for the default case.

Reply:
Looks like it is not breaking my usage